src/handlers/viewSubmitInteraction.ts

```diff
@@ -0,0 +1,76 @@
+import { IModify, IRead } from '@rocket.chat/apps-engine/definition/accessors';
+import { UIKitViewSubmitInteractionContext } from '@rocket.chat/apps-engine/definition/uikit';
+
+import { TodoistApp } from '../../TodoistApp';
+import { ModalsEnum } from '../enums/Modals';
+import { createTask } from '../lib/create/createTask';
+import { deleteComment, deleteLabel, deleteSection, deleteTask } from '../lib/delete/deleteItem';
+
+export class ExecuteViewSubmitHandler {
+  constructor(
+    private readonly app: TodoistApp,
+    private readonly read: IRead,
+    private readonly modify: IModify
+  ) {}
+
+  public async run(context: UIKitViewSubmitInteractionContext) {
+    const data = context.getInteractionData();
+    let { view, room } = data;
+    const elements = view.id.split('#');
+    const viewId = elements[0];
+    const roomId = elements[1];
+    const logger = this.app.getLogger();
+
+    if (!room) {
+      logger.warn('Room data not present in context.');
+      room = await this.read.getRoomReader().getById(roomId);
+      if (!room) {
+        logger.warn(`Room with id: ${roomId} does not exist.`);
+        return context.getInteractionResponder().errorResponse();
+      }
+    }
+
+    logger.debug(viewId, room.id);
```

Comment:
I would add some better debuging info, this is not useful at all, if it all blows app you won't even find this